### PR TITLE
Change bio-farm tech name to angels 0.7.9

### PR DESCRIPTION
**Pull request by MorMund**
The tech was renamed in  angelsbioprocessing 0.7.9 https://github.com/Arch666Angel/mods/blob/master/angelsbioprocessing/migrations/angelsbioprocessing_0.7.9.json

### CODE REVIEW

**Comment by MorMund** [comment]
I unfortunately realized that the angels update broke more than the simple tech rename. It also caused a cyclic dependency with blue algae and the chemical science pack. I am new to Factorio modding so I am not sure if my solution is optimal, but in my sea block-play through I doesn't seem to cause any deadlocks as far as I can tell. Feel free to close this if a better solution is found.

**Comment by SuperTux88** [comment]
I think the cyclic tech dependency is seablock specific, so I think that should be fixed there and not here. The seablock mod adds `bio-processing-blue` as a prerequisite for `angels-oil-processing`, so without seablock there isn't a loop.

File: tweaks/angelsmods/1_update.lua
Comment (SuperTux88) [inline tweaks/angelsmods/1_update.lua]: This isn't really needed, blue has red as dependency and red has green.

(but as said, it probably should be fixed in seablock anyway)

**Comment by MorMund** [comment]
@SuperTux88 Ah that makes sense, thank you!

**Comment by SuperTux88** [comment]
@DeadlySurprise I added this at the bottom of `SeaBlock_0.4.4/prototypes/recipe.lua` (where the dependency for `bio-processing-blue` is added too, so I guess that's the correct place):

```lua
angelsmods.functions.OV.remove_science_pack('bio-processing-blue', 'chemical-science-pack')
```

I also don't know if that's the best solution, but it solved the cyclic dependency for me.

**Comment by michaelkrymski** [comment]
Where did it change? I can't seem to find it.

**Comment by LovelySanta** [comment]
> I unfortunately realized that the angels update broke more than the simple tech rename. It also caused a cyclic dependency with blue algae and the chemical science pack. I am new to Factorio modding so I am not sure if my solution is optimal, but in my sea block-play through I doesn't seem to cause any deadlocks as far as I can tell. Feel free to close this if a better solution is found.

Blue and red algae swapped placed

**Comment by evandy0** [comment]
The blue/red algae will be an issue for trainwreck to work out, I imagine.   The tech loop is the least of it - you can't get oil to make chem science without blue algae, but blue algae is now locked behind chem science.   Trainwreck's going to have to move it back, or do something like replace blue algae with red algae in oil making.

**Comment by Cylindryk** [comment]
I don't use seablock settings, so are you sure that is the reason?

Also - how to hotfix it? I kinda don't need that bio science packs (considering it is talking about bio processing, it is about it, right?), I kinda just need to start so I can work on expanding my green and blue science...

**Comment by SuperTux88** [comment]
The blue algae change was seablock only (and that is already fixed in SeaBlock-0.4.5), the bio-farm change needs to be fixed in this mod. And if you don't want to wait until this is merged and released, you can just do the same change yourself locally and your game should start again.

**Comment by CurlyFlow** [comment]
Could someone explain what needs to be changed so we can play again? theres a long weekend coming for most germans.. :/

**Comment by michaelkrymski** [comment]
I have made a video on the hotfix for the mod: (https://www.youtube.com/watch?v=d82WMxKc-vA)
If you would like more help shoot me a dm: 
Discord; BlazingFlame#5003.

**Review by mexmer — APPROVED** [review]
quite easy fix, i will release new version with it soon. thanks

**Comment by mexmer** [comment]
also @LovelySanta yes, seems that will be the case. my pack doesn't use algae, so it should not be affected.

**Pull request merged**
